### PR TITLE
US783 Rent Manager Folder

### PR DESCRIPTION
**Pull request by dzadroga**
This US works on refactoring the Rent Manager Folder to remove all of the strategy patterns and adapters to simplify the logic and better fit the project's new architecture.  All of the Rent Calculator and Rent Integration tests now pass.

All changes have been documented in a file, that will be linked below, to show what changes were made and why. 

Refactor Explanation: [https://github.com/MissionToPsyche-Platinum/platinum_24a_game_psycheopoly-se/blob/US783/Documentation/AuditDiagrams/Rent%20Manager%20Folder/RentManagerRefactor.md](url)

Tasks Completed: 
- Remove IRentStrategy and StandardRentStrategy (Tasks 784)
- Create a static Rent Calculator (Tasks 785)
- Implement necessary help methods in rent calculator such as StreetRent, RailroadRent, and UtilityRent (Tasks 787)
- Add mortgage handling logic (Tasks 786)
- Refactor RentManager to use RentCalculator (Tasks 788)
- Remove OwnableSpaceTileAdapter and OwnershipServiceAdapter, and refactor rent logic to read directly from OwnableSpaceData (Tasks 789)
- Replace AddComponent usage with [Require] component or validation in Awake() (Tasks 790)
- Remove FindFirstObjectByType<RulesManager> if possible and wire it in the inspector  (Tasks 791)
- Upgrade Rent Event Payload to including more information: tenant, owner, tile, base rent, final rent, rent modifiers (Tasks 792)
- Create unit tests for rent calculator logic (Tasks 793)
- Make sure no compile errors or tests fila after refactor is complete (Tasks 794)
- All changes are documented in .md (Tasks 795)

### CODE REVIEW

File: Assets/Scripts/Managers/Rent/OwnershipServiceAdapter.cs [hunk 1]
```diff
@@ -12,29 +12,47 @@ public class OwnershipServiceAdapter : MonoBehaviour, IOwnershipService
...
-            if (tile is OwnableSpaceTileAdapter adapter && adapter.Data != null)
+            if (owners.TryGetValue(tile, out var previousOwner) &&
+                previousOwner != null &&
+                ownedByPlayer.TryGetValue(previousOwner, out var previousSet))
             {
-                adapter.Data.SetOwner(owner);
+                previousSet.Remove(tile);
+            }
+
+            if (tile is OwnableSpaceData ownableData)
```
Comment (hdathert) [inline Assets/Scripts/Managers/Rent/OwnershipServiceAdapter.cs:25]: This would never resolve to `true`, because `ITileRentInfo` is not derived from space data at all.

File: Assets/Scripts/Managers/Rent/RentCalculator.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+using Logging;
+using Assets.Scripts.Managers.Rules;
+
+namespace Assets.Scripts.Managers.Rent
+{
+
+    public static class RentCalculator
+    {
+        public static int ComputeRent(
+                    ITileRentInfo tile,
```
Comment (hdathert) [inline Assets/Scripts/Managers/Rent/RentCalculator.cs:12]: What does ITileRentInfo give us that we can't get from the OwnableSpaceData? We should unify our data structures so we don't have multiple versions of the same thing throughout the codebase, which would be confusing for future teams.

File: Assets/Scripts/Managers/Rent/RentManager.cs
Comment (hdathert) [inline Assets/Scripts/Managers/Rent/RentManager.cs]: Given that right now `EconomyAdapter` is only one method that is really specific to paying rent, wouldn't it just make more sense for the `RentManager` to own this method rather than having a separate component? Especially since this component is nested in a shared file (which **does not work with Unity `MonoBehaviour`, each `MonoBehaviour` needs to be its own file**).

Comment (hdathert) [inline Assets/Scripts/Managers/Rent/RentManager.cs:29]: Where is this meant to be called from? If its called automatically from a landed space, then each ownable space would need reference to this component... Which could get messy.

I'm not sure how to deal with that, but a thought I had was that since this does contain 'state' sort of, through the use of the rent modifier service, that we could make this a singleton with static access to this specific function, rather than dealing with the insane coupling that would come from having every ownable space hold reference to this. There only needs to be one RentManager at a time, so it seems like a decent candidate for the pattern.

Also this requires the `ITileRentInfo` adapter, that we had talked about moving away from since all relevant information already exists inside `OwnableSpaceData`. I'm not sure where that adaptation is meant to be happening.

**Review by hdathert — CHANGES_REQUESTED** [review]
## `OwnershipServiceAdapter`

I'm not sure I see why we need this at this point. The data that it checks on is available within the OwnableSpaceData or Player directly- the adapter is just an extra abstraction that doesn't need to be there (its a second source of 'truth', that requires manual intervention). I'd suggest we remove this entirely- I think we had talked about that in the audit, but I'm not seeing that in my notes on it, so I might have forgotten to mention it.

## `OwnableSpaceData` attributes

It seems like we've moved attributes from `ITileRentInfo` and added adapter getters into the ownable space data. While this works, it does feel like we're just band-aiding the adapter into the `OwnableSpaceData`, while the information we're looking for basically already exists inside the structure, and its giving me a code smell.

---

I think we should be retiring the `ITileRentInfo`, since its an extra layer of abstraction that does nothing for us, since all of the same data basically already exists in `OwnableSpaceData`. The addition of required attributes to `OwnableSpaceData` seems fine, but its basically just adding a separate label to existing data within it rather than addressing the problem of having different 'descriptions' of the same data.

`RentManager` needs some clarity in terms of its use. Right now, it expects to be called externally on `TryChargeRent`, but its unclear who is meant to call it. Because it is a `MonoBehavior` existing in the scene, we'd need to pass it as a reference into what ever is meant to call on it. If `PlayerController` is meant to interface with it, then it needs to hold a reference to it, as it stands. If Spaces call it, then every ownable space needs to hold reference. That could get really out of hand really fast.

If its just `PlayerControllers` calling it, thats more managable, but we'd need to put that reference into the constructor for it, and make sure that `PlayerManager` has a reference to `RentManager` for construction, which means the `GameManager` needs reference to construct the `PlayerManager`.

Another solution would be to make `RentManager` a Singleton and allow static access to the `RentManager`'s `TryChargeRent` method. That way we don't need direct references at all from anything... The downside is that coupling is more 'hidden', but with IDE finding uses of methods in the codebase, I'm a little less concerned about that than I am about the hard coupling of attaching a ton of constructor arguments, which are super brittle.

At this point I'm spitballing. If you have any questions, don't hesitate to reach out!

File: Assets/Scripts/Managers/Rent/OwnershipServiceAdapter.cs (hunk 1 above)
Comment (dzadroga) [reply Assets/Scripts/Managers/Rent/OwnershipServiceAdapter.cs:25]: This adapter is going to be removed later on due to other dependencies that are being factored out, I opened an issue in git for next sprint.

File: Assets/Scripts/Managers/Rent/RentCalculator.cs (hunk 2 above)
Comment (dzadroga) [reply Assets/Scripts/Managers/Rent/RentCalculator.cs:12]: This adapter is going to be removed later on due to other dependencies that are being factored out, I opened an issue in git for next sprint.

File: Assets/Scripts/Managers/Rent/RentManager.cs
Comment (dzadroga) [reply Assets/Scripts/Managers/Rent/RentManager.cs:29]: This adapter is going to be removed later on due to other dependencies that are being factored out, I opened an issue in git for next sprint.

**Review by hdathert — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #783 and Tasks #784 to #795  
**Code Review Completed By**: @hdathert 

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*

Some required changes will be deferred to Sprint 9 as they touch other sections of the codebase that are volatile for this sprint- issue has been logged for next sprint.

**Comment by kescob0825** [comment]
# Code Review Checklist

**Completed for**: US #783
**Code Review Completed By**: Karl Escobar

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---
